ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/WithClauseProperties.java

```diff
@@ -117,4 +119,40 @@ private static void throwOnUnknownProperty(
           + String.join(",", onlyInProvided));
     }
   }
+
+  public Optional<Character> getValueDelimiter() {
```

Comment:
`WithClauseProperties` is a generic base class for any set of properties. It should not _know_ about any specific property.  I realise that the value delimiter is present on both _current_ subclasses of this class, but that does not mean it belongs here.  If you look, both subclasses have `getValueFormat` methods.  Follow the same pattern for `getValueDelimiter`

Reply:
This would mean duplicating this method in both subclasses, is that what we really want?

Reply:
Isn't WithClauseProperties a base class for properties of all with clauses? And all with clauses can accept value delimiter?